doc/api/http2.md

```diff
@@ -0,0 +1,692 @@
+# HTTP2
+
+> Stability: 1 - Experimental
+
+The `http2` module provides an implementation of the [HTTP/2][] protocol. It
+can be accessed using:
+
+```js
+const http2 = require('http2');
+```
+
+## Core API
+
+The Core API provides a low-level interface designed specifically around
+support for HTTP/2 protocol features. It is specifically not designed with
+compatibility with the existing [HTTP/1] module API.
+
+The following illustrates a simple, plain-text HTTP/2 server:
+
+```js
+const http2 = require('http2');
+
+// Create a plain-text HTTP/2 server
+const server = http2.createServer();
+
+server.on('stream', (stream, headers) => {
+  stream.respond({
+    'content-type': 'text/html',
+    ':status': 200
+  });
+  stream.end('<h1>Hello World</h1>');
+});
+
+server.listen(80);
+```
+
+The following illustrates an HTTP/2 client:
+```
+const http2 = require('http2');
+
+const client = http2.connect('http://localhost:80');
+
+const req = client.request({ ':path': '/'});
+
+req.on('response', (headers) => {
+  console.log(headers[':status']);
+  console.log(headers['date']);
+});
+
+let data = '';
+req.setEncoding('utf8');
+req.on('data', (d) => data += d);
+req.on('end', () => client.destroy());
+req.end();
+```
+
+### Class: Http2Session
+
+* Extends: {EventEmitter}
+
+Instances of the `http2.Http2Session` class represent an active communications
+session between an HTTP/2 client and server. Instances of this class are *not*
+intended to be constructed directly by user code.
+
+Every `Http2Session` object is associated with exactly one [`net.Socket`][] or
+[`tls.TLSSocket`][]. Upon creation of the `Http2Session` instance, the `Socket`
+will be associated, with the lifecycle of each being bound to the other. When
+either is destroyed, both will be destroyed.
+
+#### Event: `'close'`
```

Comment:
Looking at the existing http docs, it doesn't look like backticks are used in the event headers (but the single quotes are).